tensorflow_lite_support/cc/test/task/vision/image_transformer_test.cc

```diff
@@ -0,0 +1,146 @@
+}
+
+class PostprocessorTest : public tflite_shims::testing::Test {};
+
+TEST_F(PostprocessorTest, FloatSucceedsWithFullMetadata) {
+  SUPPORT_ASSERT_OK_AND_ASSIGN(ImageData rgb_image, LoadImage("fox_downsampled.jpg"));
+
+  std::unique_ptr<FrameBuffer> frame_buffer = CreateFromRgbRawBuffer(
+      rgb_image.pixel_data,
+      FrameBuffer::Dimension{rgb_image.width, rgb_image.height});
+  ImageTransformerOptions options;
+  options.mutable_base_options()->mutable_model_file()->set_file_name(
+    JoinPath("./" /*test src dir*/, kTestDataDirectory,
+              kESRGANModelWithInputAndOutputMetaData));
+  SUPPORT_ASSERT_OK_AND_ASSIGN(std::unique_ptr<ImageTransformer> image_transformer,
+                      ImageTransformer::CreateFromOptions(options));
+
+  StatusOr<FrameBuffer> result_or =
+      image_transformer->Transform(*frame_buffer);
+  ImageDataFree(&rgb_image);
+  SUPPORT_ASSERT_OK(result_or);
+}
+
+TEST_F(PostprocessorTest, FloatSucceedsWithPartialMetadata) {
+  SUPPORT_ASSERT_OK_AND_ASSIGN(ImageData rgb_image, LoadImage("fox_downsampled.jpg"));
+
+  std::unique_ptr<FrameBuffer> frame_buffer = CreateFromRgbRawBuffer(
+      rgb_image.pixel_data,
+      FrameBuffer::Dimension{rgb_image.width, rgb_image.height});
+  ImageTransformerOptions options;
+  options.mutable_base_options()->mutable_model_file()->set_file_name(
+    JoinPath("./" /*test src dir*/, kTestDataDirectory,
+              kESRGANModelWithInputMetaData));
+  SUPPORT_ASSERT_OK_AND_ASSIGN(std::unique_ptr<ImageTransformer> image_transformer,
+                      ImageTransformer::CreateFromOptions(options));
+
+  StatusOr<FrameBuffer> result_or =
+      image_transformer->Transform(*frame_buffer);
+  ImageDataFree(&rgb_image);
+  SUPPORT_ASSERT_OK(result_or);
+}
+
+class SuperResolutionTest : public tflite_shims::testing::Test {};
+
+// Calculate the peak signal-to-noise ratio.
+// Original code: https://www.geeksforgeeks.org/python-peak-signal-to-noise-ratio-psnr/.
+double PSNR(const FrameBuffer& enhancedImage, const FrameBuffer& testImage) {
+  int imageSize = testImage.dimension().width * testImage.dimension().height;
+  const uint8* enhancedImagePtr = enhancedImage.plane(0).buffer;
+  const uint8* testImagePtr = testImage.plane(0).buffer;
+  double mse = 0.0;
+  for (int i = 0; i < imageSize; ++i, ++enhancedImagePtr, ++testImagePtr) {
+      mse += std::pow(static_cast<double>(*enhancedImagePtr) - static_cast<double>(*testImagePtr), 2);
+  }
+  mse /= imageSize;
+
+  double maxPixel = 255.0;
+  // Zero MSE means no noise is present in the signal.
+  double psnr = mse == 0 ? 100.0 : 20 * std::log10(maxPixel / std::sqrt(mse));
+
+  return psnr;
+}
+
+// Use a bi-cubically downsampled image as input to the model and compare
+// the model output with the original image. Expected PSNR taken from:
+// https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/image_enhancing.ipynb.
+TEST_F(SuperResolutionTest, PSNRTest) {
```

Comment:
@lu-wang-g , continuing above comment. i was talkin about this test.

Reply:
How much is your PSNR value? Is your output image clear than the input image, like is it enhanced?

Reply:
It wasn't particularly bad, around ~25. I haven't inspected the output image. How do I convert FrameBuffer to jpeg?